### DISCUSSION TITLE
safari problems

### DISCUSSION THREAD

**Discussion by FrenchFryInteractive** [pocketbase/pocketbase#3660] [Q&A]
Hi I am having some problems, with authstore in safari. Anyone else have any problems in the same manner? It does not validate.

**Comment by ganigeorgiev** [comment]
What do you mean by _"It does not validate"_? Could you elaborate and provide a code sample?

I don't have access to Mac or Safari so I don't think I'll be able to help you, but based on the user reports so far Safari has the following quirks:

- Cookie set with the `secure` flag on `localhost` doesn't work and requires HTTPS (_other browsers seems to have an exception for `localhost/127.0.0.1`_)

- `new Date(date)` constructor strictly follow the spec and will throw an error if you load a datetime string with space as delimiter instead of `T`. A workaround for this is `new Date(date.replace(' ', 'T'))`

- When authenticating with the OAuth2 all-in-one call, Safari may block popups that are initiated from async handlers (_see https://github.com/pocketbase/pocketbase/discussions/3563 and https://github.com/pocketbase/pocketbase/discussions/2429#discussioncomment-5943061_)
